pkg/services/pluginsintegration/pluginassets/pluginassets.go

```diff
@@ -69,6 +82,87 @@ func (s *Service) LoadingStrategy(_ context.Context, p pluginstore.Plugin) plugi
 	return plugins.LoadingStrategyFetch
 }
 
+// ModuleHash returns the module.js SHA256 hash for a plugin in the format expected by the browser for SRI checks.
+// The module hash is read from the plugin's MANIFEST.txt file.
+// The plugin can also be a nested plugin.
+// If the plugin is unsigned, an empty string is returned.
+// The results are cached to avoid repeated reads from the MANIFEST.txt file.
+func (s *Service) ModuleHash(ctx context.Context, p pluginstore.Plugin) string {
+	cachedValue, ok := s.moduleHashCache.Load(p.ID)
```

Comment:
I guess there's no case where a new plugin is installed during runtime (since cloud only)? Otherwise might need to include version in cache key.

Reply:
Good spot, this can also work on-prem (although behing a feature flag), so I added the version to the key in https://github.com/grafana/grafana/pull/93024/commits/44e2e81b2d7b93293e394e3389bc322f94ef1915